src/Parse/PartOfSpeech/Classes/MeishiFukushikanou.php

```diff
@@ -1,66 +1,60 @@
-     *
-     * @param array $properties
-     * @param array $previousWord
-     * @param array $previousToken
-     * @param array $currentToken
-     * @param array $nextToken
-     * @return array
      */
     public function handle(
         array $properties,
-        $previousWord,
-        $previousToken,
+        ?LimelightWord $previousWord,
+        ?array $previousToken,
         array $currentToken,
-        $nextToken
-    ) {
-        if ($nextToken) {
-            if ($nextToken['inflectionType'] === 'sahenSuru') {
-                $properties['partOfSpeech'] = 'verb';
-
-                $properties['eatNext'] = true;
-            } elseif ($this->isTokushuDaNotSahensetsuzoku($currentToken, $nextToken)) {
-                $properties['partOfSpeech'] = 'adjective';
+        ?array $nextToken
+    ): array {
+        if (!$nextToken) {
+            return $properties;
+        }
 
-                if ($nextToken['inflectionForm'] === 'taigensetsuzoku') {
-                    $properties['eatNext'] = true;
+        if ($nextToken['inflectionType'] === 'sahenSuru') {
+            $properties['partOfSpeech'] = 'verb';
 
-                    $properties['eatLemma'] = false;
-                }
-            } elseif ($nextToken['inflectionType'] === 'tokushuNai') {
-                $properties['partOfSpeech'] = 'adjective';
+            $properties['eatNext'] = true;
+        } elseif ($this->isTokushuDaDesuNotSahensetsuzoku($currentToken, $nextToken)) {
+            $properties['partOfSpeech'] = 'adjective';
 
+            if ($nextToken['inflectionForm'] === 'taigensetsuzoku') {
                 $properties['eatNext'] = true;
-            } elseif ($nextToken['partOfSpeech1'] === 'joshi' && $nextToken['literal'] === 'に') {
-                $properties['partOfSpeech'] = 'adverb';
 
-                $properties['eatNext'] = false;
+                $properties['eatLemma'] = false;
             }
+        } elseif ($nextToken['inflectionType'] === 'tokushuNai') {
+            $properties['partOfSpeech'] = 'adjective';
+
+            $properties['eatNext'] = true;
+        } elseif ($nextToken['partOfSpeech1'] === 'joshi' && $nextToken['literal'] === 'に') {
+            $properties['partOfSpeech'] = 'adverb';
+
+            $properties['eatNext'] = false;
         }
 
         return $properties;
     }
 
     /**
-     * Return true if next inflection is tokushuDa and POS2 is not
+     * Return true if next inflection is tokushuDa/Desu and POS2 is not
      * sahensetsuzoku.
-     *
-     * @param array $currentToken
-     * @param array $nextToken
-     * @return bool
      */
-    public function isTokushuDaNotSahensetsuzoku($currentToken, $nextToken)
+    public function isTokushuDaDesuNotSahensetsuzoku(array $currentToken, array $nextToken): bool
     {
-        return $nextToken['inflectionType'] = 'tokushuDa' &&
+        return ($nextToken['inflectionType'] === 'tokushuDa' || $nextToken['inflectionType'] === 'tokushuDesu') &&
```

Comment:
This one is a big change because the previous code was buggy. Not sure about the change since I'm not a Japanese expert though :sweat_smile: